src/ImageSharp/Formats/Webp/WebpChunkParsingUtils.cs

```diff
@@ -0,0 +1,351 @@
+
+            // Read all the chunks in the order they occur.
+            var info = new WebpImageInfo()
+            {
+                Width = width,
+                Height = height,
+                Features = features
+            };
+
+            return info;
+        }
+
+        /// <summary>
+        /// Reads a unsigned 24 bit integer.
+        /// </summary>
+        /// <param name="stream">The stream to read from.</param>
+        /// <param name="buffer">The buffer to store the read data into.</param>
+        /// <returns>A unsigned 24 bit integer.</returns>
+        public static uint ReadUnsignedInt24Bit(Stream stream, byte[] buffer)
+        {
+            stream.Read(buffer, 0, 3);
+            buffer[3] = 0;
+            return (uint)BinaryPrimitives.ReadInt32LittleEndian(buffer);
+        }
+
+        /// <summary>
+        /// Reads the chunk size. If Chunk Size is odd, a single padding byte will be added to the payload,
+        /// so the chunk size will be increased by 1 in those cases.
+        /// </summary>
+        /// <param name="stream">The stream to read the data from.</param>
+        /// <param name="buffer">Buffer to store the data read from the stream.</param>
+        /// <returns>The chunk size in bytes.</returns>
+        public static uint ReadChunkSize(Stream stream, byte[] buffer)
+        {
+            if (stream.Read(buffer, 0, 4) == 4)
+            {
+                uint chunkSize = BinaryPrimitives.ReadUInt32LittleEndian(buffer);
+                return (chunkSize % 2 == 0) ? chunkSize : chunkSize + 1;
+            }
+
+            throw new ImageFormatException("Invalid Webp data, could not read chunk size.");
+        }
+
+        /// <summary>
+        /// Identifies the chunk type from the chunk.
+        /// </summary>
+        /// <param name="stream">The stream to read the data from.</param>
+        /// <param name="buffer">Buffer to store the data read from the stream.</param>
+        /// <exception cref="ImageFormatException">
+        /// Thrown if the input stream is not valid.
+        /// </exception>
+        public static WebpChunkType ReadChunkType(Stream stream, byte[] buffer)
+        {
+            if (stream.Read(buffer, 0, 4) == 4)
+            {
+                var chunkType = (WebpChunkType)BinaryPrimitives.ReadUInt32BigEndian(buffer);
+                return chunkType;
+            }
+
+            throw new ImageFormatException("Invalid Webp data, could not read chunk type.");
+        }
+
+        /// <summary>
+        /// Parses optional metadata chunks. There SHOULD be at most one chunk of each type ('EXIF' and 'XMP ').
+        /// If there are more such chunks, readers MAY ignore all except the first one.
+        /// Also, a file may possibly contain both 'EXIF' and 'XMP ' chunks.
+        /// </summary>
+        public static void ParseOptionalChunks(Stream stream, WebpChunkType chunkType, ImageMetadata metadata, bool ignoreMetaData, byte[] buffer)
```

Comment:
I've just noticed we're using the abstract `Stream` type here and elsewhere. We should be using our concrete implementation.

Reply:
Changed with 2a8418f9